### ISSUE TITLE
wagtailadmin explorer to use page.specific in set_page_position

### ISSUE THREAD

**Issue by dnsl48** [wagtail/wagtail#3469]
### Issue Summary

I'm trying to override default ordering for some of my custom pages so that latest created ones would be shown first by default (`-path`).
I would also like to make explorer to show them in the same way and handle manual ordering properly for them. 

I'm currently stuck at `wagtailadmin.views.pages.set_page_position`, which doesn't take into account overridden methods of a specific page, even though some sibling functions do (e.g. `move_confirm` does).

Basically, I just need `page_to_move` to be a `specific` model so that I could override its `move` method.

That's where it all happens:
https://github.com/wagtail/wagtail/blob/stable/1.9.x/wagtail/wagtailadmin/views/pages.py#L768

### Technical details

* Wagtail version: `1.9`

**Comment by gasman** [comment]
Hi @dnsl48,

Unfortunately I think this is the kind of low-level internal detail where it's harmful for us to support customisations from external code... it's a similar situation to the `construct_explorer_page_queryset` hook ([discussion here](https://github.com/wagtail/wagtail/pull/3203#issuecomment-265503416)). Once we allow customisations - whether that's through a hook, or subclassing - that implicitly becomes part of the public API that we have to support forever afterwards.

Either we have to provide very specific guidance on the kind of customisations that are allowed in a custom `move` method (for example: changing the tree ordering is probably safe, but what about moving it to a completely different section, or moving related pages at the same time?) - which undermines the point of exposing it to the developer - or we effectively lose the ability to change the implementation of `move` within Wagtail in future releases, since we have no way of knowing which customisations we're going to break in the process.

Making the native tree ordering smarter is absolutely something we want to do in Wagtail - see #882 - and I think that really has to be something that's implemented within Wagtail, rather than "patched in" from outside. Perhaps you'd be interested in working on #882?

**Comment by dnsl48** [comment]
Fair enough. Thank you for the explanation. I'll go and look at #882 =)
